index.html

```diff
@@ -1787,6 +1942,147 @@ <h2>
+          points=]. Additionally extensions are limited to a length of 16 code
+          points.
+        </p>
+        <p>
+          In addition to complete MIME types, "*" can be used as the subtype of
+          a MIME type to match, for example, all image formats with "image/*".
+        </p>
+        <p>
+          Websites SHOULD always provide both [=MIME types=] and file
+          extensions for each accepts entry. This format is required by
+          operating systems that use [=MIME types=] to specify the [=file
+          types=]. On platforms that only use extensions, these can easily be
+          used instead.
+        </p>
+      </section>
+      <section>
+        <h3>
+          `icons` member
+        </h3>
+        <p>
+          The <a>file handler</a>'s <code><dfn data-dfn-for=
+          "file handler">icons</dfn></code> member lists images that serve as
+          graphical representations of a [=file type=] on a platform. This MAY
+          be replaced by the user agent for privacy & security reasons.
+        </p>
+      </section>
+      <section>
+        <h2>
+          Processing file handler items
+        </h2>
+        <p>
+          To <dfn>process a file handler item</dfn>, given [=ordered map=]
+          |item:ordered map|, [=URL=] |start_url:URL|, [=URL=] |scope:URL|, and
+          [=URL=] |manifest URL:URL|.
+        </p>
+        <ol class="algorithm">
+          <li>Return failure if it's the case that:
+            <ul>
+              <li>|item| is not [=ordered map=].
+              </li>
+              <li>|item|["action"] doesn't [=map/exist=] or is not a
+              [=string=].
+              </li>
+              <li>|item|["name"] doesn't [=map/exist=] or is the empty string.
+              </li>
+              <li>|item|["accepts"] doesn't [=map/exist=].
+              </li>
+              <li>|item|["accepts"] is not a [=sequence=].
+              </li>
+              <li>|item|["accepts"] is an empty [=sequence=].
+              </li>
+            </ul>
+          </li>
+          <li>Let |url:URL| be the result of [=URL Parser|parsing=]
+          |item|["action"] with |start_url URL| as the base URL.
+          </li>
+          <li>If |url| is failure, return failure.
+          </li>
+          <li>If |url| is not [=manifest/within scope=] of |scope|, return
+          failure.
+          </li>
+          <li>TODO - process "accepts"
+          </li>
+          <li>Let |file_handler:ordered map| be |ordered map| «[ "action" →
+          |url|, "name" → |item|["name"] ]».
+          </li>
+          <li>[=Process image resources=] passing |item|["icons"],
+          |file_handler|, |manifest URL|, and "icons".
+          </li>
+          <li>The user agent MAY replace |file_handler|["icons"] and
+          |file_handler|["name"] with the application icon and a string
+          containing the application name, respectively.
```

Comment:
The user agent should not replace `name` with the application's name. It may fall back to something else, such as "\<Application name\> file", but the application name by itself is a bad fallback.

Also this fallback is more intended for a missing `name`; I don't think there's a reason to ever replace the name provided by the app.